src/view/components/cpx/CpxSimulator.tsx

```diff
@@ -88,6 +85,9 @@ class Simulator extends React.Component<any, IState> {
 
     handleMessage = (event: any): void => {
         const message = event.data; // The JSON data our extension sent
+        if (message.active_device !== DEVICE_LIST_KEY.CPX) {
+            return;
+        }
```

Comment:
Does the python communication bridge run all components handleMessage? If yes, it is worth changing at the communication level rather than checking here.

Reply:
Will do. Will verify before sending set-state from vscode api